app/controllers/foreman_inventory_upload/missing_hosts_controller.rb

```diff
@@ -0,0 +1,24 @@
+module ForemanInventoryUpload
+  class MissingHostsController < ::ApplicationController
+    def index
+      organizations = User.current.my_organizations
+      organization_id = organizations.pluck(:id)
+      payload = InsightsMissingHosts.where(organization_id: organization_id)
+
+      render :json => payload
+    end
+
+    def remove_hosts
+      organization_id = params[:organization_id]
+      search_term = params[:search_term]
+
+      task = ForemanTasks.sync_task(ForemanInventoryUpload::Async::RemoveInsightsHostsJob, search_term, organization_id)
```

Comment:
```suggestion
      task = ForemanTasks.async_task(ForemanInventoryUpload::Async::RemoveInsightsHostsJob, search_term, organization_id)
```
The user doesn't have to wait for the action to finish.